File: app/src/test/java/com/coinninja/coinkeeper/util/currency/BTCtoUSDConversionsTest.java
package com.coinninja.coinkeeper.util.currency;

import org.junit.Before;
import org.junit.Test;

import static org.hamcrest.CoreMatchers.equalTo;
import static org.hamcrest.MatcherAssert.assertThat;

public class BTCtoUSDConversionsTest {

    private final static double CONVERSION_VALUE = 6603.30D;
    private Currency evalueationCurrency;

    @Before
    public void setUp() {
        evalueationCurrency = new USDCurrency(CONVERSION_VALUE);
    }

    /**
     * Given USD: 0
     * Given BTC: .01976285
     */
    @Test
    public void convertsBTCtoUSD() {
        BTCCurrency btcCurrency = new BTCCurrency("0.01976285");

        USDCurrency usdCurrency = btcCurrency.toUSD(evalueationCurrency);

        assertThat(usdCurrency.toFormattedCurrency(), equalTo("$130.50"));
    }

    /**
     * Given USD: 130
     * Given BTC: 0
     */
    @Test
    public void convertsUSDtoBTC() {
        USDCurrency usdCurrency = new USDCurrency("130.5");

        BTCCurrency btcCurrency = usdCurrency.toBTC(evalueationCurrency);

        assertThat(btcCurrency.toLong(), equalTo(1976285L));
    }


    /**
     * eval usd 6612.70
     * Given USD: 3
     * Given BTC: .00045367
     */
    @Test
    public void roundingIssueUSDtoBTC() {
        USDCurrency eval = new USDCurrency(6612.7D);
        BTCCurrency btcCurrency = new BTCCurrency(".00045367");

        USDCurrency usdCurrency = btcCurrency.toUSD(eval);

        assertThat(usdCurrency.toFormattedCurrency(), equalTo("$3.00"));
    }
}

File: common/src/test/java/app/dropbit/commons/currency/BTCtoUSDConversionsTest.kt
package app.dropbit.commons.currency

import org.hamcrest.CoreMatchers.equalTo
import org.junit.Assert.assertThat
import org.junit.Test

class BTCtoUSDConversionsTest {
    private val evaluationCurrency: Currency = USDCurrency(6603_30)

    /**
     * Given USD: 0
     * Given BTC: .01976285
     */
    @Test
    fun convertsBTCtoUSD() {
        val btcCurrency = BTCCurrency(0.01976285)
        val usdCurrency = btcCurrency.toUSD(evaluationCurrency)
        assertThat(usdCurrency.toFormattedCurrency(), equalTo("$130.50"))
    }

    /**
     * Given USD: 130
     * Given BTC: 0
     */
    @Test
    fun convertsUSDtoBTC() {
        val usdCurrency = USDCurrency(130.5)
        val btcCurrency = usdCurrency.toBTC(evaluationCurrency)
        assertThat(btcCurrency.toLong(), equalTo(1976285L))
    }

    /**
     * eval usd 6612.70
     * Given USD: 3
     * Given BTC: .00045367
     */
    @Test
    fun roundingIssueUSDtoBTC() {
        val eval = USDCurrency(6612.7)
        val btcCurrency = BTCCurrency(.00045367)
        val usdCurrency = btcCurrency.toUSD(eval)
        assertThat(usdCurrency.toFormattedCurrency(), equalTo("$3.00"))
    }
}